x-pack/elastic-agent/pkg/agent/cmd/diagnostics.go

```diff
@@ -361,6 +363,35 @@ func gatherConfig() (AgentConfig, error) {
 	}
 	cfg.ConfigRendered = mapCFG
 
+	// Gather vars to render process config
+	isStandalone, err := isStandalone(renderedCFG)
+	if err != nil {
+		return cfg, err
```

Comment:
[Nit / personal preference]
Usually I rather return a newly created empty struct to make explicit that the returned value is useless. Besides `cfg` might have some data, which I believe, should not be used by the caller when an error happens.
```suggestion
		return AgentConfig{}, err
```